### PR TITLE
Added type definitions for leaflet-responsive-popup

### PR DESCRIPTION
**Pull request by stankurek**
Please fill in this template.

- [ v] Use a meaningful title for the pull request. Include the name of the package modified.
- [ v] Test the change in your own code. (Compile and run.)
- [ v] Add or edit tests to reflect the change. (Run with `npm test`.)
- [v ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [v ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ v] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [v ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--

### CODE REVIEW

File: types/leaflet-responsive-popup/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (sheetalkamat) [inline types/leaflet-responsive-popup/tsconfig.json]: strictness cannot be turned off, please turn on both flags and fix errors if any

File: types/leaflet-responsive-popup/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// TypeScript Version: 2.3
+
+import * as L from 'leaflet';
+
+declare module 'leaflet' {
+    interface ResponsivePopupOptions extends PopupOptions {
+        hasTip?: boolean;
+    }
+
+    class responsivePopup extends Popup {
```
Comment (sheetalkamat) [inline types/leaflet-responsive-popup/index.d.ts]: Looking at https://unpkg.com/leaflet-responsive-popup@0.6.4/leaflet.responsive.popup.js::
`ResponsivePopup` seems like a class and `responsivePopup` seems like a function so you need both, one class and one function

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/leaflet-responsive-popup/tsconfig.json (hunk 1 above)
Comment (stankurek) [reply types/leaflet-responsive-popup/tsconfig.json]: Thank you for your review. I changed the file.

File: types/leaflet-responsive-popup/index.d.ts (hunk 2 above)
Comment (stankurek) [reply types/leaflet-responsive-popup/index.d.ts]: Good point. Thanks for taking the time to check it.

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**